internal/pgdb/pgdb_index.go

```diff
@@ -55,25 +59,42 @@ func (module *Module) extraIndexes(ctx pgsgo.Context, m pgs.Message, ix *importT
 		return rv
 	}
 
-	rv.DB.Method = idx.Nethod
+	rv.DB.Method = idx.Method
+
 	tenantIdField, err := getTenantIDField(m)
 	if err != nil {
 		panic(err)
 	}
+
 	for _, fieldName := range idx.Columns {
-		f := fieldByName(m, fieldName)
-		rv.SourceFields = append(rv.SourceFields, ctx.Name(f).String())
-
-		if fieldName == tenantIdField {
-			rv.DB.Columns = append(rv.DB.Columns, "tenant_id")
-		} else {
-			pgColName, err := getColumnName(f)
-			if err != nil {
-				panic(err)
+		path := strings.Split(fieldName, "🌮")
+		message := m
+		resolution := ""
+		for i, p := range path {
+			f := fieldByName(message, p)
+			if i != len(path)-1 {
+				resolution += getNestedName(f)
+				message = f.Type().Embed()
+				continue
+			}
+
+			if p == tenantIdField {
```

Comment:
This is the wrong tenantIdField.  We can not actually get the correct one here given the limited API.